ushahidi/platform-client-mzima
# Ush 1190 - Meaningful error messages on mobile around offline posts being uploaded

**Issue:** 

When a user creates a post in offline mode, the platform displays a success message indicating the post has been created.However, upon regaining an internet connection, the platform displays a toast error message..

**Changes:**

This PR changes the error messages around offline posts being uploaded to be meaningful to users. It also removes a few redundant calls to the API, when the app is aware there is no network, that was causing excessive error messages.

**Testing:**

1. Open the mobile app and activate airplane mode (or disable wifi/mobile data) to simulate offline mode.
2. Start creating a new survey post and fill out all the required information.
3. Submit the completed survey post.
4. Observe the success message displayed by the platform indicating the post creation.
5. Deactivate airplane mode (or reconnect to wifi/mobile data) to regain internet connection.
6. Wait a few moments for the app to attempt syncing offline data.
7. Look for a toast error message appearing on the screen mentioning the upload failure.
8. confirm the post is created using other device
